### PR TITLE
feat(quantic): quantic smart snippet answer created

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-4797](https://coveord.atlassian.net/browse/SFINT-4797
)
- The Quantic Smart Snippet Answer component created.
- The Quantic Smart Snippet Answer displays the answer provided as an HTML markup.
- The Quantic Smart Snippet Answer binds the correct actions to the events triggered  on the links within  this component in order to log analytics.

#### Design doc:
[Design-doc__Quantic-Smart-Snippet-components](https://docs.google.com/document/d/1kJswWu6qHp5ExMHQagkrCWphrjRmys8a5tQ0oAnQkJU/edit)

#### Demo:

AnswerDemo

[SFINT-4797]: https://coveord.atlassian.net/browse/SFINT-4797?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetAnswer/quanticSmartSnippetAnswer.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
+import {LinkUtils} from 'c/quanticUtils';
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticSmartSnippetAnswer/quanticSmartSnippetAnswer.js:1]: is this in another branch?

Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippetAnswer/quanticSmartSnippetAnswer.js:1]: yes, it's here: https://github.com/coveo/ui-kit/pull/2707

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetAnswer/quanticSmartSnippetAnswer.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  renderedCallback() {
+    this.updateSmartSnippetAnswer();
+    this.isInitialRender = false;
+  }
+
+  updateSmartSnippetAnswer() {
+    const snippetAnswerElement = this.template.querySelector(
+      '.smart-snippet-answer'
+    );
+    // eslint-disable-next-line @lwc/lwc/no-inner-html
```
Comment (SimonMilord) [inline packages/quantic/force-app/main/default/lwc/quanticSmartSnippetAnswer/quanticSmartSnippetAnswer.js:49]: Do we want eslint to be disabled here?

**Review by SimonMilord — APPROVED** [review]
👍

**Review by lbergeron — APPROVED** [review]
Well done! It looks all good!

**Review by nathanlb — APPROVED** [review]
Nice

File: packages/quantic/force-app/main/default/lwc/quanticSmartSnippetAnswer/quanticSmartSnippetAnswer.js (hunk 2 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticSmartSnippetAnswer/quanticSmartSnippetAnswer.js:59]: yes, this eslint rule is usually useful cause we don't wanna modify HTML from the Js bundle of our LWC but in this particular case we are using the `lwc:dom="manual"`  directive in the HTML template so we can safely use `innerHTML`.

**Pull request merged**